rpc-tests/index.js

```diff
@@ -128,22 +128,32 @@ async function checkMessages(api, exp, programs) {
   return errors;
 }
 
-async function checkMemory(api, exp) {
+async function checkMemory(api, exp, programs) {
   const errors = [];
 
   for (const mem of exp.memory) {
-    if (mem.kind === 'shared') {
-      const gearMemoryOpt = await api.rpc.state.getStorage('g::memory');
-      const gearMemory = gearMemoryOpt.unwrap().toU8a();
-      const at = parseInt(mem.at, 16) - (256 * 65536);
-      const bytes = Uint8Array.from(Buffer.from(mem.bytes.slice(2), 'hex'));
-      for (let index = at; index < at + bytes.length; index++) {
-        if (gearMemory[index] !== bytes[index - at]) {
-          errors.push("Memory doesn't match");
-          break;
+
+    const gearProgramOpt = await api.rpc.state.getStorage(`0x${Buffer.from('g::prog::').toString('hex')}${programs[mem.program_id].slice(2)}`);
+    const gearProgram = api.createType('Program', gearProgramOpt.unwrap());
+
+    let at = parseInt(mem.at, 16);
+    console.log(at)
```

Comment:
Better to make the output more informative. Or drop it if it was introduced when debugging only.